### ISSUE TITLE
Able to use root / as base instead of /dashboard

### ISSUE THREAD

**Issue by azrin1972** [FlowFuse/node-red-dashboard#586]
### Description

in Dashboard 1.0, if we set in settings.js - ui : { path: "" }, we can  use https://{URL}/. as the bas of our dashboard pages. This is very useful when we need to hide the development page by setting httpAdminRoot to for example "/admin".

It would be good if we can do the same as all my dashboards are using the same concept

Thank you 

### Have you provided an initial effort estimate for this issue?

I am no FlowFuse team member

**Comment by joepavitt** [comment]
This is something I had hoped for in the early stages of Dashboard 2.0, but is annoyingly tricky to get right. Definitely want this capability in at some point, but it's unlikely to be an immediate priority

**Comment by jflemingsignifi** [comment]
@joepavitt is it possible to get dashboards working under httpAdminRoot?  If a node red instance is set up under https://{URL}/node-red/ for example dashboards are just not working at all.

**Comment by jflemingsignifi** [comment]
@joepavitt turns out the dashboards are working under a sub path when httpRoot is set rather than httpAdminRoot

**Comment by joepavitt** [comment]
> @joepavitt turns out the dashboards are working under a sub path when httpRoot is set rather than httpAdminRoot

As you've found, the `httpAdminRoot` just defines where the Node-RED Editor runs. 

`httpNodeRoot` defines where any HTTP endpoints (all `http-in` nodes and Dashboard nodes) run.

**Comment by AndreKR** [comment]
Is this the issue that tracks the development of this option?

![image](https://github.com/FlowFuse/node-red-dashboard/assets/1188538/c9f63514-ada8-493a-8e9f-1eb9e59aff02)

**Comment by rcarmo** [comment]
This is a major blocker for me right now. Is it being worked on at all?

**Comment by joepavitt** [comment]
Not currently, could you expand on the need and use-case for this please?

**Comment by rcarmo** [comment]
Yes, of course. It is a primary requirement for migrating a few dozen production Node-RED dashboards from various systems at an industrial customer. All of them follow the same URL patterns:

- / for the dashboard itself (which is a composite UI with charts and parameters for the application)
- /admin for logging in
- /api for remote monitoring endpoints
- /static for any static assets (there is a sub-folder under that where CSV files are generated)

All of these are behind a reverse proxy that maps hostnames to internal IP addresses and allows requests to go through to the production floor. There are a few other asset paths that need to go through, obviously, but the primary customer requirement is to be able to access the dashboards by virtual hostname alone and no sub-paths, and given that most of the desktops are locked down even changing bookmarks in browsers to add "/dashboard" is a major chore.

**Comment by AndreKR** [comment]
@rcarmo It's a whole different thing but FYI I'm currently using ThingsBoard for dashboards because it does have the ability to serve multiple dashboards from one instance.

**Comment by AndreKR** [comment]
@rcarmo I'm not affiliated with ThingsBoard in any way, I don't even particularly like it (setting up the rule engine is rather complicated and the available dashboard controls are rather inflexible). I just wanted to let you know that because of this issue I have tried a lot of different options to serve multiple dashboards and ThingsBoard is the one I had the least amount of trouble with. I'm sorry if the information was useless to you.

**Comment by colinl** [comment]
> All of these are behind a reverse proxy that maps hostnames to internal IP addresses

Can the reverse proxy map, for example, the domain root to <ip>:1880/dashboard?

**Comment by soleous** [comment]
Do we have a fix for this yet?  

I'm using httpNodeRoot for api endpoints (/api) and I'd like the dashboards to be ether root or path "/ui/" like the original node-red dashboard not /api/dashboard/.

**Comment by bartbutenaers** [comment]
@soleous,
As you can see this issue is still open, so it is obviously not solved.  There are lots of other open issues, so the (very small) team is currently overwhelmed.  And since nearly nobody in our community wants to contribute; you will have to be patient like every body else.  Or if it is really urgent for you: you can try to find the root cause and fix it via a pull request.  Because that is the whole idea of open-source...
Bart

**Comment by rcarmo** [comment]
> > All of these are behind a reverse proxy that maps hostnames to internal IP addresses
> 
> Can the reverse proxy map, for example, the domain root to `<ip>:1880/dashboard`?

No, I made that plain in the rest of the description. That is not an option.

**Comment by soleous** [comment]
> @soleous, As you can see this issue is still open, so it is obviously not solved. There are lots of other open issues, so the (very small) team is currently overwhelmed. And since nearly nobody in our community wants to contribute; you will have to be patient like every body else. Or if it is really urgent for you: you can try to find the root cause and fix it via a pull request. Because that is the whole idea of open-source... Bart

I'll be honest, I wasn't expecting such a nice and welcoming response.  Its the type of response that will encourage contribution... so I'll be right on that!

**Comment by rcarmo** [comment]
Any updates on whether this is being worked on? Would like to start 2025 by planning conversion of the 1.0 deployments... and this is still a blocker.

**Comment by Leonel-Toledo** [comment]
Greetings,

  First to all, thank you for the development of this solution.

  The relevance of this feature seems to be underrated for development team. Like original poster (and me), lots of people require to be able to login to the system by IP or URL, without the need of a folder. This way of work was most likely already implemented with node-red-dashboard, and it's difficult to try to explain to all users that IP or URL is not enough to access a system.

  My project relies heavily in frontend, and even when I already migrated all flows to use dashboard-V2, I have to keep using V1 just for the inability to access dashboard as homage (www.mysite.com)

**Comment by bartbutenaers** [comment]
> The relevance of this feature seems to be underrated for development team

@Leonel-Toledo 
No this feature is not underrated at all.  Nor is any other of the 349 other issues underrated.  Every issue is important for the user that registered it.  Just like this issue is important for you...

There is currently no *"development team"* like you call it.  At the moment it is mostly only me that is creating pull requests.  The Flowfuse team has been so kind to develop this dashboard for free for us and deliver it as open source, but currently they are busy creating other open-source stuff for us (non-dashboard related). 

Anybody from our community can contribute to this repo.  But if nobody is interested to help, not much will happen like you see...

**Comment by rafageist** [comment]
While Dashboard 2.0 does not yet support the same path configuration as 1.0, a workaround is to serve the dashboard at the root / on the main domain while keeping the admin interface on a dedicated subdomain, using a reverse proxy like [Caddy](https://caddyserver.com/). This ensures that pages, WebSocket connections, and other resources work correctly.

```bash
# Dashboard
mydomain.com {
    encode gzip
    header {
        Strict-Transport-Security "max-age=31536000;"
    }

    @root path /
    redir @root /home 302

    handle {
        rewrite * /dashboard{uri}
        reverse_proxy localhost:1880
    }
}

# Admin
admin.mydomain.com {
    encode gzip
    header {
        Strict-Transport-Security "max-age=31536000;"
    }

    reverse_proxy localhost:1880
}
```

An example of this setup is live at [paramaribo.info](https://paramaribo.info/) (work in progress).

**Comment by rcarmo** [comment]
Bumping this since the reverse proxy workarounds are not feasible for our use cases.

**Comment by avsd** [comment]
@rafageist 's suggestion with `rewrite` and `redir` works initially, but stops working whenever you refresh, because the `/` is cached. You need a full refresh or clearing user data for it to work. This is a blocker for me when installing it to a mobile home screen!

**Comment by rafageist** [comment]
@avsd Is true. I honestly prefer a native, elegant solution, not workarounds. I think the project needs to improve it.
